Makefile

```diff
@@ -36,10 +36,17 @@ TESTFLAGS    :=
 DUPLFLAGS    := -t 100
 
 ifeq ($(STATIC),1)
+# Static linking with glibc is a bad time. If we're building statically,
+# assume musl is installed and use that.
```

Comment:
Linux and OSX are not the only operating systems. But they're the only ones we support right now, so I'll leave that aside.

More to the point, a musl-based static build is an attractive nuisance even for linux users. People *think* they need/want a fully-static build (just like we did!), but in they almost certainly don't. A musl-based build has different semantics from a standard glibc build (hostname resolution and anything else using nsswitch.conf works differently). 

We can capture this knowledge somewhere without putting it in our main makefile which carries the implication that it is a first-class build target (and that our regular build is "dynamic" in ways that you might want to avoid).